lib/chef/knife/bootstrap/client_builder.rb

```diff
@@ -80,9 +80,12 @@ def node_name
           knife_config[:chef_node_name]
         end
 
-        # @return [String] enviroment from the knife_config
+        # @return [String] environment from the knife_config or first_boot_attributes, if none specified by knife_config
         def environment
-          knife_config[:environment]
+          # prefer environment from cli over environment from first-boot attributes
+          knife_config[:environment] or
```

Comment:
`and` and `or` are control flow operators, whereas `&&` and `||` are logical (boolean) operators. See: http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/